File: 219-ContainsDuplicateII/ContainsDuplicateII.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        if k == 0:
            return False
        j = set()
        for i in range(0, len(nums)):
            if nums[i] in j:
                return True
            else:
                if i >= k:
                    j.discard(nums[i - k])
            j.add(nums[i])
        return False

File: 219-ContainsDuplicateII/ContainsDuplicateII.cpp
#include <unordered_set>
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        if (k == 0) {
            return false;
        }
        unordered_set<int> j;
        for (int i = 0; i < nums.size(); i++) {
            if (j.find(nums[i]) != j.end()) {
                return true;
            }
            else {
                if (i >= k) {
                    j.erase(nums[i - k]);
                }
            }
            j.insert(nums[i]);
        }
        return false;
    }
};